zenoh/src/publication.rs

```diff
@@ -117,6 +121,22 @@ impl PutBuilder<'_, '_> {
         self.kind = kind;
         self
     }
+
+    #[zenoh_macros::unstable]
+    pub fn attachments(&self) -> Option<&Attachments> {
+        self.attachments.as_ref()
+    }
+
+    #[zenoh_macros::unstable]
+    pub fn attachments_mut(&mut self) -> &mut Option<Attachments> {
```

Comment:
`attachments` returns `Option<&Attachments>`, `attachments_mut` should return `Option<&mut Attachments>`.

Reply:
With the mutable accessor, I wanted to give the user the opportunity to straight up remove the attachments, so a reference to the whole option is needed